jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```
@@ -728,14 +733,35 @@ protected void initializeDefaultsComplete()
     {
     }
 
+    /**
+     * <p>Enters the scope of the {@link Context}.</p>
+     * <p> Attempts to set the {@link Thread#setContextClassLoader(ClassLoader) thread's context classloader}
+     * to the {@link #getClassLoader() configured one} if a non-null one was set.</p>
+     *
+     * @param contextRequest the context's {@link Request}
+     * @return the configured classloader if it was successfully set as the thread's context classloader,
+     * {@code null} otherwise.
+     */
     protected ClassLoader enterScope(Request contextRequest)
     {
+        CURRENT_CONTEXT.set(_context);
+
         ClassLoader lastLoader = Thread.currentThread().getContextClassLoader();
-        __context.set(_context);
         if (_classLoader != null)
-            Thread.currentThread().setContextClassLoader(_classLoader);
+        {
+            try
+            {
+                Thread.currentThread().setContextClassLoader(_classLoader);
+            }
+            catch (Throwable x)
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("error setting a context classloader on thread {}", Thread.currentThread(), x);
+                lastLoader = null;
+            }
```

We should fail here rather than ignore